spring-cloud-dataflow-docs/src/main/asciidoc/streams.adoc

```diff
@@ -23,8 +23,96 @@ http --server.port=8091 | file --directory=/tmp/httpdata/
+
+```
+source.http=maven://org.springframework.cloud.stream.app:http-source-rabbit:1.0.0.BUILD-SNAPSHOT
+sink.log=maven://org.springframework.cloud.stream.app:log-sink-rabbit:1.0.0.BUILD-SNAPSHOT
+```
+
+Then to import the apps in bulk, use the `app import` command and provide the location of the properties file via `--uri`:
+
+```
+dataflow:>app import --uri file:///<YOUR_FILE_LOCATION>/stream-apps.properties
+```
+
+You can also pass the `--local` option (which is TRUE by default) to indicate whether the
+properties file location should be resolved within the shell process itself. If the location should
+be resolved from the Data Flow Server process, specify `--local false`.
+
+When using either `app register` or `app import`, if a stream app is already registered with
+the provided name and type, it will not be overridden by default. If you would like to override the
+pre-existing stream app, then include the `--force` option.
+
+[NOTE]
+In some cases the Resource is resolved on the server side, whereas in others the
+URI will be passed to a runtime container instance where it is resolved. Consult
+the specific documentation of each Data Flow Server for more detail.
+
+[[spring-cloud-dataflow-stream-app-whitelisting]]
+=== Whitelisting application properties
+
+Stream applications are Spring Boot applications which are aware of many link:http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/#common-application-properties[common application properties], e.g. `server.port` but also families of properties such as those with the prefix `spring.jmx` and `logging`.  When creating your own application it is desirable to whitelist properties so that the shell and the UI can display them first as primary properties when presenting options via TAB completion or in drop-down boxes.
+
+To whitelist application properties create a file named `spring-configuration-metadata-whitelist.propreties` in the `META-INF` resource directory.  There are two property keys that can be used inside this file. The first key is named `configuration-properties.classes`.  The value is a comma separated list of fully qualified `@ConfigurationProperty` class names.  The second key is `configuration-properties.names` whose value is a comma separated list of property names.  This can contain the full name of propety, such as `server.port` or a partial name to whitelist a category of property names, e.g. `spring.jmx`.
+
+The link:https://github.com/spring-cloud/spring-cloud-stream-app-starters[Spring Cloud Stream application starters] are a good place to look for examples of usage.  Here is a simle example of the file source's `spring-configuration-metadata-whitelist.properties` file
+
+```
+configuration.classes=org.springframework.cloud.stream.app.file.sink.FileSinkProperties
+```
+
+If for some reason we also wanted to add `server.port` to this file, it would look like
```

Comment:
let's say file.prefix instead of server.port.  Its more closely inline with file.